plugins/coral-plugin-offtopic/client/components/OffTopicCheckbox.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import styles from './styles.css';
+import {addCommentTag} from 'coral-framework/graphql/mutations';
+import {compose} from 'react-apollo';
+
+class OffTopicCheckbox extends React.Component {
+
+  handleChange = (e) => {
+    if (e.target.checked) {
+      this.props.addCommentHooks({
+        postSubmit: {
+          addTag: (data) => {
+            const {comment} = data.createComment;
+            this.props.addCommentTag({
+              id: comment.id,
+              tag: 'OFF_TOPIC',
+            });
+          }
+        }
+      })
+    } else {
+      this.props.addCommentHooks({
+        postSubmit: {
+          addTag: () => {}
+        }
+      })
+    }
+  }
+
+  render() {
+    return (
+      <div className={styles.offTopic}>
+        <label>
```

Comment:
I wouldn't call it pollution, more like information about your intent.
It makes it easier for someone else to look at it and knows what it does.
Also a developer expects that Components are isolated, so if someone
would modify this class and adds another Component into the render, he will 
be surprised why all of his labels have a different styling.